From 3d98bbf884a7effe64179b3dafc552189577819a Mon Sep 17 00:00:00 2001
From: Ilya Dryomov <ilya.dryomov@inktank.com>
Date: Fri, 27 Jun 2014 21:46:33 +0400
Subject: [PATCH 137/213] rbd: do not leak image_id in rbd_dev_v2_parent_info()
References: fate#318918
Git-commit: fbba11b3bec52ff560cb42d102f61341049defb0
Patch-mainline: v3.17-rc1

image_id is leaked if the parent happens to have been recorded already.
Fix it.

Signed-off-by: Ilya Dryomov <ilya.dryomov@inktank.com>
Reviewed-by: Alex Elder <elder@linaro.org>
Acked-by: David Disseldorp <ddiss@suse.de>

---
 drivers/block/rbd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/block/rbd.c b/drivers/block/rbd.c
index e6921d6..420d015 100644
--- a/drivers/block/rbd.c
+++ b/drivers/block/rbd.c
@@ -4126,6 +4126,8 @@ static int rbd_dev_v2_parent_info(struct rbd_device *rbd_dev)
 		parent_spec->snap_id = snap_id;
 		rbd_dev->parent_spec = parent_spec;
 		parent_spec = NULL;	/* rbd_dev now owns this */
+	} else {
+		kfree(image_id);
 	}
 
 	/*
-- 
2.1.4

